print-bgp.c

```diff
@@ -2543,6 +2559,80 @@ bgp_attr_print(netdissect_options *ndo,
             len -= 12;
         }
         break;
+    case BGPTYPE_BGPSEC_PATH:
+    {
+        uint16_t sblen, splen, siglen;
+
+        splen = GET_BE_U_2(tptr);
+
+        /*
+         * A secure path has a minimum length of 8 bytes:
+         * 2 bytes length field
+         * 6 bytes per secure path segment
+         */
+        if (splen < 8) {
+            ND_PRINT(" [secure path too short]");
+            break;
+        }
+
+        ND_PRINT("\n\t    Secure Path Length: %u", splen);
+
+        tptr += 2;
+        splen -= 2;
+        /* Make sure that there are secure path segments present */
+        if (splen == 0 || splen % 6) {
+            ND_PRINT(" [invalid total segments len %u]", splen);
+            break;
+        }
+        /* Parse secure path segments */
+        while (splen != 0) {
+            uint8_t pcount = GET_U_1(tptr);
+            uint8_t flags = GET_U_1(tptr + 1);
+            uint32_t asn = GET_BE_U_4(tptr + 2);
+            ND_PRINT("\n\t      Secure Path Segment: pCount: %u, Flags: [%s] (0x%02x), AS: %u",
+                     pcount,
+                     bittok2str(bgp_bgpsec_bitmap_str, "none", flags),
+                     flags,
+                     asn);
+            tptr += 6;
+            splen -= 6;
+        }
+
+        sblen = GET_BE_U_2(tptr);
+
+        /*
+         * A signature block has a minimum length of 95 bytes:
+         * 2 bytes length field
+         * 1 bytes algorithm id
+         * 92-94 bytes per signature segment
+         *  '-> 20 bytes SKI, 2 bytes length field, 70-72 bytes signature
```

Comment:
To better understand how to test for length errors:
"70-72 bytes signature". Where is this specified?

Reply:
I was trying really hard to actually find a resource that states how long a ECDSA signature is. I solely base this claim on my observations of never having seen a signature of different size other than 70, 71 or 72 bytes. Not the most solid foundation to stand on, I must confess...

Reply:
>  I solely base this claim on my observations of never having seen a signature of different size other than 70, 71 or 72 bytes.

**If** the answer to the following question is correct (and **if** I understand it correctly), it would only be a matter of probability.
https://bitcoin.stackexchange.com/questions/112662/what-is-the-probability-of-an-ecdsa-signature-being-less-than-71-bytes

Reply:
Ah I see, highly interesting! I didn't new this. I guess the check for a minimum length must be set to a lower value. Should it be 64+3, according to the SE answer? Or even lower, in case a signature could potentially be even smaller in size?

Reply:
As stated in the answer "Its output contains:": It's only a partial output with the highest probabilities.